openedx/core/djangoapps/content/search/management/commands/reindex_studio.py

```diff
@@ -0,0 +1,137 @@
+        with self.using_temp_index(index_name) as temp_index_name:
+            ############## Configure the index ##############
+            # Mark usage_key as unique (it's not the primary key for the index, but nevertheless must be unique):
+            client.index(temp_index_name).update_distinct_attribute(Fields.usage_key)
+            # Mark which attributes can be used for filtering/faceted search:
+            client.index(temp_index_name).update_filterable_attributes([
+                Fields.block_type,
+                Fields.context_key,
+                Fields.org,
+                Fields.tags,
+                Fields.type,
+            ])
+            # Mark which attributes are used for keyword search, in order of importance:
+            client.index(temp_index_name).update_searchable_attributes([
+                Fields.display_name,
+                Fields.block_id,
+                Fields.content,
+                Fields.tags,
+                # Keyword search does _not_ search the course name, course ID, breadcrumbs, block type, or other fields.
+            ])
+
+            ############## Libraries ##############
+            self.stdout.write("Indexing libraries...")
+            for lib_key in lib_keys:
+                self.stdout.write(f"{num_contexts_done + 1}/{num_contexts}. Now indexing library {lib_key}")
+                docs = []
+                for component in lib_api.get_library_components(lib_key):
+                    metadata = lib_api.LibraryXBlockMetadata.from_component(lib_key, component)
+                    doc = searchable_doc_for_library_block(metadata)
+                    docs.append(doc)
+                    num_blocks_done += 1
+                # Add all the docs in this library at once (usually faster than adding one at a time):
+                self.wait_for_meili_task(client.index(temp_index_name).add_documents(docs))
+                num_contexts_done += 1
+
+            ############## Courses ##############
+            self.stdout.write("Indexing courses...")
+            for course in all_courses:
+                self.stdout.write(
+                    f"{num_contexts_done + 1}/{num_contexts}. Now indexing course {course.display_name} ({course.id})"
+                )
+                docs = []
+
+                def add_with_children(block):
+                    """ Recursively index the given XBlock/component """
+                    doc = searchable_doc_for_course_block(block)
+                    docs.append(doc)  # pylint: disable=cell-var-from-loop
+                    self.recurse_children(block, add_with_children)  # pylint: disable=cell-var-from-loop
+
+                self.recurse_children(course, add_with_children)
+
+                # Add all the docs in this course at once (usually faster than adding one at a time):
+                self.wait_for_meili_task(client.index(temp_index_name).add_documents(docs))
```

Comment:
I'm not sure if this could happen here, but an additional check doesn't hurt.
```suggestion
                if docs:
                    # Add all the docs in this course at once (usually faster than adding one at a time):
                    self.wait_for_meili_task(client.index(temp_index_name).add_documents(docs))
```